x-pack/legacy/plugins/snapshot_restore/server/routes/api/recovery.test.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Request, ResponseToolkit } from 'hapi';
+import { getAllHandler } from './recovery';
+
+describe('[Snapshot and Restore API Routes] Recovery', () => {
+  const mockRequest = {} as Request;
+  const mockResponseToolkit = {} as ResponseToolkit;
+  const mockEsShard = {
+    type: 'SNAPSHOT',
+    source: {},
+    target: {},
+    index: { size: {}, files: {} },
+    translog: {},
+  };
+
+  describe('getAllHandler()', () => {
+    it('should arrify and filter recovery shards returned from ES', async () => {
+      const mockEsResponse = {
+        fooIndex: {
+          shards: [mockEsShard],
+        },
+        barIndex: {
+          shards: [mockEsShard, mockEsShard],
+        },
+        testIndex: {
+          shards: [
+            {
+              ...mockEsShard,
+              type: 'EMPTY_STORE',
+            },
+          ],
+        },
+      };
+      const callWithRequest = jest.fn().mockReturnValueOnce(mockEsResponse);
+      const expectedResponse = [
+        {
+          index: 'fooIndex',
+          shards: [{}],
+          isComplete: false,
+          latestActivityTimeInMillis: 0,
+        },
+        {
+          index: 'barIndex',
+          shards: [{}, {}],
```

Comment:
This seems strange... from the deserialization and endpoint logic, it looks like these shards should contain some of the ES data, e.g. `index: { size: {}, files: {} }`. Do you know why we expect them to be empty objects?

Reply:
the serialization logic strips out undefined settings _after_ destructuring (aka `index: { size: {}, files: {} }` is essentially flattened, then stripped out)